pycaret/internal/plots/time_series.py

```diff
@@ -988,6 +1002,26 @@ def plot_time_series_differences(
             "plot_time_series_differences() only works on a single time series, "
             f"but {data.shape[1]} series were provided."
         )
+
+    title = "Difference Plot"
+    if model_name is not None:
+        title = f"{title} | Model Residuals"
+        column_titles = [f"{model_name[0]} Residuals"]
+    elif data_label is not None:
+        title = f"{title} | {data_label}"
+        # Do not use data_label for column titles since the actual column name
+        # has the more detailed data_type (e.g. with "transformed" at the end.)
+        column_titles = [data.columns[0]]
+    else:
+        # Both model_name and data_label are None
+        raise ValueError(
+            "Both model_name and data_label can not be None. Please specify one based "
+            "on where the results are coming from (model or data respectively)."
+            "\nIf you believe that this error should not be raised, please file an issue "
+            "on GitHub with a reproducible example:"
+            "\nhttps://github.com/pycaret/pycaret/issues"
```

Comment:
Should we make this exception message a variable, as it's being reused?

Reply:
change made